src/metabase/upload.clj

```diff
@@ -145,11 +146,24 @@
                 (t/plus (t/seconds 1))
                 (t/truncate-to :seconds))))))
 
+(def ^:private transliterator (Transliterator/getInstance "Any-Latin; Latin-ASCII"))
+
+(defn- transliterate [s]
+  (when-not (str/blank? s)
+    (-> (.transliterate ^Transliterator transliterator ^String s)
+        (str/replace #"\s+" "_")
+        (str/replace #"[^\w]+" "_")
+        (str/replace #"_{2,}" "_")
+        (str/replace #"^_|_$" ""))))
+
 (defn- unique-table-name
   "Append the current datetime to the given name to create a unique table name. The resulting name will be short enough for the given driver (truncating the supplied `table-name` if necessary)."
   [driver table-name]
+  ;; TODO we should not rely on the timestamp to make the filename unique
+  ;; Ideally we would add an incrementing count, but it may be cheaper and easier to include some randomness.
   (let [time-format                 "_yyyyMMddHHmmss"
-        slugified-name               (or (u/slugify table-name) "")
+        slugified-name               (transliterate table-name)
+        slugified-name               (if (str/blank? slugified-name) "blank" slugified-name)
```

Comment:
```suggestion
        slugified-name               (or (not-empty (transliterate table-name)) "blank")
```

maybe like this? 😁

Reply:
Calling seq on a string is a bit gross, wastes memory, but this is much cleaner! Didn't think of it.

Reply:
we really need `str/not-blank` <_<